twitter_ads/resource.py

```diff
@@ -42,12 +43,17 @@ def __init__(self, account):
     def account(self):
         return self._account
 
-    def from_response(self, response):
+    def from_response(self, response, headers=None):
         """
         Populates a given objects attributes from a parsed JSON API response.
         This helper handles all necessary type coercions as it assigns
         attribute values.
         """
+        if headers is not None:
```

Comment:
What happens for the case where headers is `None` and the calling function i.e., `return self.from_response(response.body['data'], response.headers)` has the `response.headers` set? Ideally this should have something like `cursor.account_rate_limit` set to `None` as well.

Reply:
@tushdante Thanks! That's a good point... I removed explicit "if" conditions from the `extract_response_headers()` and changed to use `dict.get()` method so they get either actual value from response or `None` in case if it's not present in response, and always accessible through instance variable.